### PR TITLE
Cut 5114 reg import update data type

### PR DESCRIPTION
**Pull request by jworkmanjc**
## Issues
* [CUT-5114](https://jumpcloud.atlassian.net/browse/CUT-5114) - Update .reg file import for JCPolicy

## What does this solve?

This release addresses an issue with the JumpCloud Registry Policy for Windows devices. Previously when creating a new policy from a registry .reg file, the New-JCPolicy function would call our private function `Convert-RegToPSObject.ps1` and in doing so would import `int` type DWORDS as `int` type values. This change explicitly casts these values as `string` type values. 

When these fields were cast as `int` types, the JumpCloud Agent would fail to apply the policy based on it's own validation of the registry entry. Now when creating policies DWORD type values are cast as strings regardless of their type and they are able to successfully deployed to devices. 

## Is there anything particularly tricky?

N/A

## How should this be tested?

Save this file as a .reg file somewhere on your desktop:
```reg
Windows Registry Editor Version 5.00

[HKEY_LOCAL_MACHINE\SOFTWARE\Policies\Microsoft\WindowsFirewall\DomainProfile]
"EnableFirewall"=dword:00000000
```

Using the latest production version of the JumpCloud PowerShell Module, create a new policy using this .reg file and `New-JCPolicy`

`New-JCPolicy -TemplateName windows_Advanced:_Custom_Registry_Keys -RegistryFile "~/Documents/registry.reg" -Name "(Testing) Windows Firewall Policy"`

This should create a new policy in JumpCloud, assign this policy to a windows device, it should fail.

Error on JumpCloud Console:
Screenshot 2026-05-06 at 3 14 23 PM

Error on `jcagent.log`
Screenshot 2026-05-06 at 3 16 47 PM

In a new PWSH session, import the development version of the PowerShell module:
`Import-Module "~/Documents/GitHub/support/PowerShell/JumpCloud Module/JumpCloud.psd1" -force`

Create the policy from the same file using a different name:
`New-JCPolicy -TemplateName windows_Advanced:_Custom_Registry_Keys -RegistryFile "~/Documents/registry.reg" -Name "(Testing) Windows Firewall Policy -latest"`

Apply that policy to the same device, it should now apply correctly:
Screenshot 2026-05-06 at 3 19 43 PM

## Screenshots

In previous versions of `Convert-RegToPSObject.ps1` some registry values would fail to apply and show this error in the `jcagent.log` file:
image

[CUT-5114]: https://jumpcloud.atlassian.net/browse/CUT-5114?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

---

> [!NOTE]
> **Medium Risk**
> Changes the registry `.reg` import serialization for `New-JCPolicy`/`Set-JCPolicy`, which can affect how Windows registry policies are applied and may alter existing expectations around numeric types/edge cases (e.g., zero and 64-bit ranges). Test updates reduce regression risk but behavior changes are user-visible.
> 
> **Overview**
> Fixes Windows custom registry policy creation from `.reg` files by changing `Convert-RegToPSObject` to emit **DWORD and QWORD values as decimal strings** (using `uint32`/`uint64` parsing) instead of PowerShell integers, and adds safer handling for all-zero QWORD inputs.
> 
> Updates the Pester tests to assert the new string-typed `customData` for DWORD/QWORD, and refreshes the module manifest/changelog release dates to reflect the cut.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 5995a875fbc749ae793c8c72eab6492af9a589fa. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: PowerShell/JumpCloud Module/JumpCloud.psd1 [hunk 1]
```diff
@@ -3,7 +3,7 @@
 #
 # Generated by: JumpCloud Solutions Architect Team
 #
-# Generated on: 5/01/2026
+# Generated on: 5/56/2026
```
[the remark this answers is not in the document]
Comment (jworkmanjc) [reply PowerShell/JumpCloud Module/JumpCloud.psd1]: resolved in cea143ff9c72951f30be93e2334ec229ea291da5

**Review by junioralmeida-82 — APPROVED** [review]
Approved. Verified the 4 files changed and they correctly address the alias fixes and documentation updates for the v3.1.0 release. Great work on the rebase.
